addons/microsoft_calendar/models/calendar.py

```diff
@@ -74,13 +74,47 @@ def create(self, vals_list):
         if self._check_microsoft_sync_status() and not notify_context and recurrency_in_batch:
             self._forbid_recurrence_creation()
 
-        # for a recurrent event, we do not create events separately but we directly
-        # create the recurrency from the corresponding calendar.recurrence.
-        # That's why, events from a recurrency have their `need_sync_m` attribute set to False.
-        return super(Meeting, self.with_context(dont_notify=notify_context)).create([
-            dict(vals, need_sync_m=False) if vals.get('recurrence_id') or vals.get('recurrency') else vals
-            for vals in vals_list
-        ])
+        # Create events separately since the sender user may differ between events.
+        events = self.env['calendar.event']
+        users_by_id = self._get_users_by_id(vals_list)
+        vals_by_user = {}
+        for vals in vals_list:
+            # If event has a different organizer, check its sync status and verify if the user is listed as attendee.
+            sender_user, partner_ids = self._get_organizer_user_change_info(vals, users_by_id)
+            partner_included = partner_ids and len(partner_ids) > 0 and sender_user.partner_id.id in partner_ids
+            self._check_organizer_validation(sender_user, partner_included)
+            # Group new events values by user for later batch creation.
+            if sender_user in vals_by_user:
+                vals_by_user[sender_user].append(vals)
+            else:
+                vals_by_user[sender_user] = [vals]
+
+        for sender_user in vals_by_user:
+            # for a recurrent event, we do not create events separately but we directly
+            # create the recurrency from the corresponding calendar.recurrence.
+            # That's why, events from a recurrency have their `need_sync_m` attribute set to False.
+            events |= super(Meeting, self.with_user(sender_user).with_context(dont_notify=notify_context)).create([
+                dict(vals, need_sync_m=False) if vals.get('recurrence_id') or vals.get('recurrency') else vals
```

Comment:
maybe you can set need_sync_m in the loop above so you don't need to do this list comprehension here